cmd/cli/user.go

```diff
@@ -75,3 +79,88 @@ func deleteUser(database moira.Database, user string) error {
 
 	return nil
 }
+
+func handleCleanup(logger moira.Logger, database moira.Database, config cleanupConfig) error {
+	var users []string
+
+	reader := json.NewDecoder(bufio.NewReader(os.Stdin))
+
+	if err := reader.Decode(&users); err != nil {
+		return err
+	}
+
+	return usersCleanup(logger, database, users, config)
+}
+
+func usersCleanup(logger moira.Logger, database moira.Database, users []string, config cleanupConfig) error {
+	usersMap := make(map[string]bool, len(users)+len(config.Whitelist))
+
+	if config.HandlingAnonymous {
+		config.Whitelist = append(config.Whitelist, "")
+	}
+
+	for _, user := range append(users, config.Whitelist...) {
+		usersMap[user] = true
+	}
+
+	contacts, err := database.GetAllContacts()
+	if err != nil {
+		return err
+	}
+
+	if len(contacts) == 0 {
+		return nil
+	}
+
+	var usersNotFound []string
+
+	for _, contact := range contacts {
+		if contact == nil {
+			continue
+		}
+
+		if !usersMap[contact.User] {
+			usersNotFound = append(usersNotFound, contact.User)
+		}
+	}
+
+	for _, user := range usersNotFound {
+		if config.Delete {
+			if err = deleteUser(database, user); err != nil {
+				break
+			}
+		}
+
+		if !config.Delete {
+			if err = offNotification(database, user); err != nil {
+				break
+			}
+		}
```

Comment:
Instead of `if` here you can place `continue` in previous `if`. But probably it is better to extract this functionality to another function that handles user.

Reply:
corrected